crates/bevy_asset/src/assets.rs

```diff
@@ -561,9 +559,10 @@ impl<A: Asset> Assets<A> {
                         continue;
                     }
                 }
-                remove_asset = infos.process_handle_drop(untyped_id);
+                asset_unused = infos.process_handle_drop(untyped_id);
```

Comment:
Minor nit: instead of a mutable `asset_unused`, have this `if` statement return line 562, and `true` on the `else` branch.